hydrolib/core/io/ini/models.py

```diff
@@ -153,74 +48,126 @@ class Comments(BaseModel, ABC):
             arbitrary_types_allowed = False
 
     comments: Optional[Comments] = None
 
-    @validator("comments", always=True)
+    @validator("comments", always=True, allow_reuse=True)
     def comments_matches_has_comments(cls, v):
-        if cls._supports_comments() and v is None:
-            raise ValueError(f"{cls} should have comments.")
-        elif not cls._supports_comments() and v is not None:
-            raise ValueError(f"{cls} should not have comments.")
-
+        if not cls._supports_comments() and v is not None:
+            logging.warning(f"Dropped unsupported comments from {cls.__name__} init.")
+            v = None
         return v
 
     @classmethod
     def validate(cls: Type["IniBasedModel"], value: Any) -> "IniBasedModel":
         if isinstance(value, Section):
-            # value = value.flatten()  # TODO
-            value = cls._convert_section(value)
+            value = value.flatten(
+                cls._duplicate_keys_as_list(), cls._supports_comments()
+            )
 
         return super().validate(value)
 
-    @classmethod
-    def _convert_section(cls, section: Section) -> Dict:
-        converted_content = cls._convert_section_content(section.content)
-        underlying_dict = cls._convert_section_to_dict(section)
-        return {**underlying_dict, **converted_content}
+
+class DataBlockIniBasedModel(IniBasedModel):
+    """DataBlockIniBasedModel defines the base model for ini models with datablocks."""
 
     @classmethod
     def _convert_section_to_dict(cls, value: Section) -> Dict:
         return value.dict(
             exclude={
                 "start_line",
                 "end_line",
-                "datablock",
                 "content",
             }
         )
 
+
+class INIGeneral(IniBasedModel):
+    fileVersion: str = "3.00"
+    fileType: str
+
     @classmethod
-    def _convert_section_content(cls, content: List) -> Dict:
-        def group_and_flatten(l: Iterable[Tuple[str, Any]]) -> Dict[str, Any]:
-            if cls._duplicate_keys_as_list():
-                return reduce(_combine_in_dict, l, {})
-            else:
-                return dict(l)
-
-        values = group_and_flatten(
-            (v.key, v.value) for v in content if isinstance(v, Property)
-        )
+    def _supports_comments(cls):
+        return False
 
-        if cls._supports_comments():
-            values["comments"] = group_and_flatten(
-                (v.key, v.comment) for v in content if isinstance(v, Property)
-            )
 
-        return values
+class CrossdefGeneral(INIGeneral):
+    fileVersion: str = "3.00"
+    fileType: Literal["crossDef"] = "crossDef"
 
 
-class DataBlockIniBasedModel(IniBasedModel):
-    """DataBlockIniBasedModel defines the base model for ini models with datablocks."""
+class CrosslockGeneral(INIGeneral):
+    fileVersion: str = "3.00"
+    fileType: Literal["crossLoc"] = "crossLoc"
+
+
+class INIModel(FileModel):
```

Comment:
I'm not sure if I understand this class. Is it a model that corresponds with a single ini file? If so, then the doc string does not fit? If the docstring is correct, then shouldn't it use a `.mdu` extension. And shouldn't it need an additional step after the `Parser.parse`?

Reply:
Yeah, single `.ini` model. I've updated the docstring.